pandas/io/formats/style.py

```diff
@@ -1222,6 +1222,76 @@ class MyStyler(cls):
 
         return MyStyler
 
+    def pipe(self, func, *args, **kwargs):
+        """
+        Apply func(self, \*args, \*\*kwargs), and return the result.
+
+        .. versionadded:: 0.24.0
+
+        Parameters
+        ----------
+        func : function
+            function to apply to the Styler.
+            ``args``, and ``kwargs`` are passed into ``func``.
+            Alternatively a ``(callable, data_keyword)`` tuple where
+            ``data_keyword`` is a string indicating the keyword of
+            ``callable`` that expects the Styler.
+        args : iterable, optional
+            positional arguments passed into ``func``.
+        kwargs : mapping, optional
+            a dictionary of keyword arguments passed into ``func``.
+
+        Returns
+        -------
+        object : the value returned by ``func``.
+
+        See Also
+        --------
+        Styler.apply
+        Styler.applymap
+        pandas.DataFrame.pipe
+
+        Notes
```

Comment:
I think this should be the summary more than a Notes section.

Reply:
I'm inclined to leave this as "notes", for consistency with [pandas.DataFrame.pipe](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.pipe.html).  

Another reason to separate this from the function specification is that this section alludes to specific use cases for `pipe`, and conventions for the user-defined functions.  But I wouldn't want users to get the impression that they must use `pipe` like this -- there may be several other useful applications for the method that creative users will devise.

Reply:
We'll chage `DataFrame.pipe` then (you can open yourself another PR for it). We should show this in the examples (no code blocks should be in any other section). And the code needs to show real-world examples, and needs to be runnable.

Reply:
This was added at the request of @gfyoung [earlier in the review process](https://github.com/pandas-dev/pandas/pull/23384#discussion_r228727737), who asked:

> Also, for what purpose?
> 
> (yes, as reviewers, we can see what the rationale was from your PR, so just put that rationale into the docstring)

@TomAugspurger [suggested making this a code block](https://github.com/pandas-dev/pandas/pull/23384#issuecomment-435075033).

I think removing the code block may increase compliance with some developer rules for docstrings, but would make "rationale" part of the docstring less clear to end users.

@gfyoung, do you think the text-only rationale, without the code examples, is sufficient?

Reply:
I like the code block here.

Reply:
Not a big deal, but I don't think the code block will be run in the doctests, or validated for pep8 in the docstring validation. That's why I would prefer to have the code in the example, and not in code blocks (besides a personal preference on keeping all the code together).

Reply:
This code block is more for showing why `pipe` is still useful despite being functionally equivalent to function calls. It's not intended as an actual example.